### ISSUE TITLE
Bug: Handling errors with the NuxtErrorBoundary component

### ISSUE THREAD

**Issue by lautiamkok** [nuxt/framework#9321]
The `NuxtErrorBoundary` component does not seem to work as [expected](https://nuxt.com/docs/getting-started/error-handling#rendering-errors-within-your-app):

> The #error slot will receive error as a prop. (If you set error = null it will trigger re-rendering the default slot; 

Example:

```
// app.vue
<template>
  <NuxtErrorBoundary>
    <NuxtPage/>
    <template #error="{ error }">
      <p>An error occurred: {{ error }}</p>
      <button @click="() => clearError({ redirect: '/' })">
        This will clear the error.
      </button>
    </template>
  </NuxtErrorBoundary>
</template>
```

**Problem 1:**

The error is cleared at all. The error message still can be seen on the screen.

**Problem 2:**

The Home page is not rendered at all. It is still rendering the error page and message.

Any ideas?

**Comment by manniL** [comment]
Hey! Could you kindly provide a [reproduction](https://v3.nuxtjs.org/community/reporting-bugs/#create-a-minimal-reproduction) via StackBlitz or CodeSandbox? 🙏

**Comment by lautiamkok** [comment]
@manniL here it is:

[https://stackblitz.com/edit/github-qsnf12?file=app.vue](https://stackblitz.com/edit/github-qsnf12?file=app.vue)

the `NuxtErrorBoundary` component does not work on StackBlitz at all. So you need to download the sample app and run it locally then you see the problems.

Thanks

**Comment by manniL** [comment]
Does it work fine with using it *inside* a page (so not `app.vue`)?

**Comment by lautiamkok** [comment]
@manniL nope. if you put that error component on `/pages/about.vue`, you will get the following error on your Console:

```
[vue-router warn]: uncaught error during route navigation:
[vue-router.js?v=414e8b35:2490 TypeError: num.toUpperCase is not a function
```

**Comment by sangyoo91** [comment]
Trying to use `NuxtErrorBoundary` on a component/page level, and I dont think its preventing global error page from rendering. I'm throwing createError from API and trying to gracefully show it while not rendering the global error page. Hopefully this ifxes soon... I'm about to pull all my hair out.

**Comment by metkm** [comment]
Yep, Even though I use `NuxtErrorBoundary` the full error page shows up.

**Comment by lautiamkok** [comment]
If you use the plain Vue 3 to handle the exact scenario, it works correctly and perfectly. So, if you want your Vue app to handle errors correctly, Nuxt 3 is still not a choice to go for. Btw, now you can use Vue 3 with Vite alone or a Vite plugin to create your own Vue SSR app. So if you can wait for this error to be fixed in Nuxt, create your own Vue SSR apps.

**Comment by CharlesBT** [comment]
Hi I have the same issue using NuxtErrorBoundary around NuxtPage 👍 
    <NuxtErrorBoundary>
        <NuxtPage />
    </NuxtErrorBoundary>

Any idea or workaround ?
